finder-app/finder.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+# finder script for ECEN5713 AESD Assignment 1
+#Author: Justin Denning
+
+set -e
+set -u
+
+#Verify we got both of the required arguments
+if [ $# -lt 2 ]; then 
+	echo "Script usage: finder.sh filesdir searchstr"
+	echo -e "\tfilesdir - the directory to search for files in"
+	echo -e "\tsearchstr - the string to find"
+	echo -e "\nThis script will first count all the files in filesdir (including subdirectories), then it will search the contents of those files for searchstr and return the number of matching lines"
+	exit 1
+fi
```

Comment:
@JLDenn a suggestion here: after getting by the initial argument checks I'd save the $1 and $2 arguments into more verbose/readable variable names.